katsdpcontroller/scheduler.py

```diff
@@ -2307,15 +2340,21 @@ def run_in_event_loop(func, *args, **kw):
     task_id = request.match_info['id']
     task, graph = scheduler.get_task(task_id, return_graph=True)
     if task is None:
-        return aiohttp.web.HTTPNotFound(reason='Task ID {} not active\n'.format(task_id))
+        raise aiohttp.web.HTTPNotFound(text='Task ID {} not active\n'.format(task_id))
     else:
         try:
             for dep in task.depends_ready:
                 await dep.ready_event.wait()
+                if not dep.was_ready:
+                    logger.info('Not starting %s because %s died', task.name, dep.name)
+                    raise aiohttp.web.HTTPServiceUnavailable(
+                        text=f'Dependency {dep.name} died without becoming ready\n')
+        except (asyncio.CancelledError, aiohttp.web.HTTPException):
+            raise
         except Exception as error:
             logger.exception('Exception while waiting for dependencies')
-            return aiohttp.web.HTTPInternalServerError(
-                reason='Exception while waiting for dependencies:\n{}\n'.format(error))
+            raise aiohttp.web.HTTPInternalServerError(
```

Comment:
Does this get logged again by an outer handler ? We don't want the same error logged twice.

Reply:
The aiohttp logging will log the request URL and the 500 status, but it won't show the error message. And the container will print the error to stderr, but I don't think it'll generally get from there into our logging system.